### PR TITLE
Input list sorted based on time stamp

### PR DESCRIPTION
**Pull request by AliceBalfanz**
This PR implements sorting of the input list based on the dataset time stamps by default. The order of the input file list may be kept by passing `--no_sort`. The parameter `--sort` is deprecated with this PR.

### CODE REVIEW

File: docs/source/cli/xcube_gen.rst [hunk 1]
```diff
@@ -68,10 +68,14 @@ Generate xcube dataset.
                                       slices.
       --prof                          Collect profiling information and dump
                                       results after processing.
-      --sort                          The input file list will be sorted before
-                                      creating the xcube dataset. If --sort
-                                      parameter is not passed, order of input list
-                                      will be kept.
+      --no_sort                       The input file list wont be sorted before
+                                      creating the xcube dataset. If --no_sort
+                                      parameter is passed, order the input list
```
Comment (dzelge) [inline docs/source/cli/xcube_gen.rst]:
```suggestion
parameter is passed, the order of the input list
```

File: xcube/cli/gen.py [hunk 2]
```diff
@@ -66,9 +66,11 @@
               help='Deprecated. The command will now always create, insert, replace, or append input slices.')
 @click.option('--prof', is_flag=True,
               help='Collect profiling information and dump results after processing.')
-@click.option('--sort', is_flag=True,
-              help='The input file list will be sorted before creating the xcube dataset. '
-                   'If --sort parameter is not passed, order of input list will be kept.')
+@click.option('--no_sort', is_flag=True,
+              help='The input file list wont be sorted before creating the xcube dataset. '
```
Comment (dzelge) [inline xcube/cli/gen.py]:
```suggestion
              help='The input file list will not be sorted before creating the xcube dataset. '
```

File: xcube/cli/gen.py [hunk 3]
```diff
@@ -66,9 +66,11 @@
...
+                   'If --no_sort parameter is passed, order the input list will be kept. '
```
Comment (dzelge) [inline xcube/cli/gen.py]:
```suggestion
                   'If --no_sort parameter is passed, the order of the input list will be kept. '
```

File: CHANGES.md [hunk 4]
```diff
@@ -2,6 +2,12 @@
 
 ### New
 
+* Added new parameter in `xcube gen` called `--no_sort`. Using `--no_sort`, 
+the input file list wont be sorted before creating the xcube dataset. 
+If `--no_sort` parameter is passed, order the input list will be kept. 
+The parameter `--sort` is deprecated and the input files will be sorted 
+by default.
```
Comment (forman) [inline CHANGES.md]: Indent

File: docs/source/cli/xcube_gen.rst [hunk 5]
```diff
@@ -68,10 +68,14 @@ Generate xcube dataset.
                                       slices.
       --prof                          Collect profiling information and dump
                                       results after processing.
-      --sort                          The input file list will be sorted before
-                                      creating the xcube dataset. If --sort
-                                      parameter is not passed, order of input list
-                                      will be kept.
+      --no_sort                       The input file list wont be sorted before
```
Comment (forman) [inline docs/source/cli/xcube_gen.rst]:
```suggestion
      --no_sort                       The input file list will not be sorted before
```

**Pull request merged**